week-4-kuberflow-pipelines/README.md

```diff
@@ -0,0 +1,240 @@
+replicaset.apps/ml-pipeline-persistenceagent-bd9f8d4d7       1         1         1       23m
+replicaset.apps/ml-pipeline-scheduledworkflow-544c8bbc58     1         1         1       23m
+replicaset.apps/ml-pipeline-ui-6c895bb85b                    1         1         1       23m
+replicaset.apps/ml-pipeline-viewer-crd-79db74f698            1         1         1       23m
+replicaset.apps/ml-pipeline-visualizationserver-74fbc54649   1         1         1       22m
+replicaset.apps/mysql-67f7987d45                             1         1         1       22m
+replicaset.apps/proxy-agent-bff474798                        1         1         0       22m
+replicaset.apps/workflow-controller-594fd96fd5               1         1         1       22m
+
+NAME                              TYPE                 VERSION   OWNER   READY   AGE
+application.app.k8s.io/pipeline   Kubeflow Pipelines   1.8.5     true            23m
+```
+## Public URL
+
+Get the public URL for the Kubeflow Pipelines UI and use it to access the Kubeflow Pipelines UI:
+
+```bash
+kubectl get services --namespace kubeflow
+```
+
+```
+NAME                              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
+cache-server                      ClusterIP   10.96.134.166   <none>        443/TCP             23m
+controller-manager-service        ClusterIP   10.96.139.135   <none>        443/TCP             23m
+metadata-envoy-service            ClusterIP   10.96.141.158   <none>        9090/TCP            23m
+metadata-grpc-service             ClusterIP   10.96.205.99    <none>        8080/TCP            23m
+minio-service                     ClusterIP   10.96.178.254   <none>        9000/TCP            23m
+ml-pipeline                       ClusterIP   10.96.241.67    <none>        8888/TCP,8887/TCP   23m
+ml-pipeline-ui                    ClusterIP   10.96.188.166   <none>        80/TCP              23m
+ml-pipeline-visualizationserver   ClusterIP   10.96.27.184    <none>        8888/TCP            23m
+mysql                             ClusterIP   10.96.236.138   <none>        3306/TCP            23m
+workflow-controller-metrics       ClusterIP   10.96.199.168   <none>        9090/TCP            23m
+```
+
+So, we see that `ml-pipeline-ui` is on port 80:
+
+```bash
+kubectl port-forward service/ml-pipeline-ui 8000:80 -n kubeflow
+```
+
+At the moment on the web UI `localhost:8000` you'll see `Error: failed to retrieve list of pipelines.` error, because there are no pipelines created yet.
+
+
+Delete local cluster after you're done:
+
+```bash
+kubectl delete all --all --all-namespaces
+kind delete cluster --name=local-cluster
+```
+
+## Training on the Local Kubernetes Cluster
+
+### Build and push Dataset Downloader Docker Image
+
+```bash
+cd load-data
+docker build --rm -t datadownloader .
+docker tag datadownloader:latest yevhenk10s/datadownloader:latest
+docker push yevhenk10s/datadownloader:latest
+```
```

Comment:
should we use CI/CD for this?

Reply:
Definitely.  In case of any changes of the data set downloader it is not good idea to manually rebuild its container.